src/plugins/expressions/common/executor/executor.ts

```diff
@@ -192,10 +195,12 @@ export class Executor<Context extends Record<string, unknown> = Record<string, u
     Output = unknown
   >(
     ast: string | ExpressionAstExpression,
-    context: ExtraContext = {} as ExtraContext
+    context: ExtraContext = {} as ExtraContext,
+    functionCache?: Map<string, any>
```

Comment:
we don't use it anywhere yet, but in dashboard for example there might be multiple instances of executor (visualize using core executor, but lens embeddable using its forked version. so it might be nice if we are able to provide same cache to all of them. but i can remove this from now.